music21/metadata/__init__.py

```diff
@@ -308,48 +743,281 @@ def all(self, skipContributors=False):
 
         return list(sorted(allOut.items()))
 
+    def _getStringValueByNSKey(self, nsKey: str) -> t.Optional[str]:
+        values: t.Tuple[ValueType, ...]
+        try:
+            values = self._get(nsKey, isCustom=False)
+        except KeyError:
+            return None
+
+        if not values:
+            return None
+
+        if self._isContributorNSKey(nsKey):
+            if len(values) == 1:
+                return str(values[0])
+            if len(values) == 2:
+                return str(values[0]) + ' and ' + str(values[1])
+            return str(values[0]) + f' and {len(values)-1} others'
+
+        if self._needsArticleNormalization(nsKey):
+            output: str = ''
+            for i, value in enumerate(values):
+                assert isinstance(value, Text)
+                if i > 0:
+                    output += ', '
+                output += value.getNormalizedArticle()
+            return output
+
+        return ', '.join(str(value) for value in values)
+
+    def _getStringValuesByNSKey(self, nsKey: str) -> t.List[str]:
+        values: t.Tuple[ValueType, ...]
+        try:
+            values = self._get(nsKey, isCustom=False)
+        except KeyError:
+            return []
+
+        if not values:
+            return []
+
+        if self._needsArticleNormalization(nsKey):
+            output: t.List[str] = []
+            for value in values:
+                assert isinstance(value, Text)
+                output.append(value.getNormalizedArticle())
+            return output
+
+        return [str(value) for value in values]
+
+    def _getPluralAttribute(self, attributeName: str) -> t.List[str]:
+        # This does what __getattr__ would do if we supported plural attributeNames
+        # (but it takes singular attributeNames, of course).
+
+        # It raises AttributeError if attributeName is not a valid uniqueName,
+        # workId, or workId abbreviation.  Used in search, for example, because
+        # search wants to find everything.
```

Comment:
this has docs, needs examples.

Reply:
Can you point me at a good example of how to do this?

Reply:
How to do "they probably need to be explicitly put in DOC_ATTR", that is.